### PR TITLE
11 get articles topic query

### PR DESCRIPTION
**Pull request by Veresio**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -105,6 +105,24 @@ describe("GET /api/articles", () => {
         expect(body.articles).toBeSortedBy("created_at", { descending: true });
       });
   });
+  test("should return only the articles that have the correct topics according to the query", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        body.articles.forEach((article) => {
```
Comment (roisinoreilly) [inline __tests__/server.test.js:114]: don't forget to do a length check to avoid a false positive on this test

File: __tests__/server.test.js [hunk 2]
```diff
@@ -105,6 +105,24 @@ describe("GET /api/articles", () => {
...
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("should return a 404 error if there are no articles for that topic", () => {
```
Comment (roisinoreilly) [inline __tests__/server.test.js]: this test description isn't really what you're testing, you're checking here that a valid but non-existent topic gives you a 404 which is correct, just update the description to be a bit clearer

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
good job so far! I'd like you to update the test description for your 404 test and also add in a 400 test for an invalid topic eg. /articles?topic=12345. Once you've done that send us another PR :)

File: __tests__/server.test.js [hunk 3]
```diff
@@ -105,6 +105,34 @@ describe("GET /api/articles", () => {
         expect(body.articles).toBeSortedBy("created_at", { descending: true });
       });
   });
+  test("should return only the articles that have the correct topics according to the query", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBeGreaterThan(0);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/server.test.js:113]: You can be a bit more specific here, based on the test data how many articles should there be in the response ? - that is the length we would expect

File: __tests__/server.test.js [hunk 4]
```diff
@@ -105,6 +105,34 @@ describe("GET /api/articles", () => {
...
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("should return a 404 error if we're being passed a valid topic with no articles", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("No articles available with that topic");
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/server.test.js:124]: If the topic _exists_ in the database then this isn't a 404, it's a `200`. The response should just be an empty array, similar to if there is an article with no comments from ticket 6

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Looks good, I've left two bits of feedback below. The second one is a fairly important change. Once you've made changes this can be merged to main, no need for another review ✅

**Pull request merged**